examples/model_selection/plot_nested_cross_validation_iris.py

```diff
@@ -0,0 +1,119 @@
+"""
+=========================================
+Nested versus Non-Nested cross-validation
+=========================================
+
+This example compares non-nested and nested cross-validation strategies on a
+classifier of the iris data set. Nested cross-validation (CV) is often used to
+train a model in which hyperparameters also need to be optimized. Nested CV
+approximates the generalization error of the resulting estimator of the
+hyperparameter search. It is generally best practice to use nested CV as
+non-nested CV will sometimes provide a slightly more biased and optimistic
+score.
+
+In contrast to non-nested CV, an inner CV loop is introduced that partitions
```

Comment:
How about: 

Model selection without nesting CV involves evaluating the model's performance on data that is also used to tune the model. Information may thus "leak" into the model and overfit the data. The magnitude of this effect is primarily dependent on the size of the dataset and the stability of the model. See Cawley and Talbot [1]_ for an analysis of these issues.

Nested CV effectively uses a series of train/validation/test set splits. Score is approximately maximised in fitting a model to each training set, and then directly maximised in selecting hyperparameters over the validation set. Assessing performance on a held-out test set avoids evaluating a model on data that has been used to tune it. Generalization error is estimated by averaging test set scores over several dataset splits.